service/dps/storm/common/src/main/java/eu/europeana/cloud/service/dps/storm/NotificationBolt.java

```diff
@@ -124,14 +124,25 @@ private void notifyTask(NotificationTuple notificationTuple, NotificationCache n
         int processesFilesCount = nCache.getProcessed();
         int errors = nCache.getErrors();
 
+        //notification table (for single record)
         storeNotification(processesFilesCount, taskId,
                 notificationTuple.getParameters());
 
         if (error) {
             storeNotificationError(taskId, nCache, notificationTuple);
         }
-        if ((processesFilesCount % PROCESSED_INTERVAL) == 0) {
+        if(isCounterUpdateRequired(nCache)){
+            LOGGER.info("Updating task counter for task_id = {} and counter value: {}", taskId, processesFilesCount);
             taskInfoDAO.setUpdateProcessedFiles(taskId, processesFilesCount, errors);
+            nCache.setLastCounterUpdate(new Date());
+        }
+    }
+
+    private boolean isCounterUpdateRequired(NotificationCache nCache) {
+        if (new Date().getTime() - nCache.getLastCounterUpdate().getTime() > COUNTER_UPDATE_INTERVAL_IN_MS) {
+            return true;
+        } else {
+            return false;
         }
```

Comment:
This could be replaced by just one line if you return the if-condition directly.

Reply:
Yes, I will change it.